apps/core/app/[locale]/(default)/cart/_components/shipping-info.tsx

```diff
@@ -0,0 +1,227 @@
+  const t = useTranslations('Cart.SubmitShippingInfo');
+
+  return (
+    <Button className="w-full items-center px-8 py-2" disabled={pending} variant="secondary">
+      {pending ? (
+        <>
+          <Spinner aria-hidden="true" className="animate-spin" />
+          <span className="sr-only">{t('spinnerText')}</span>
+        </>
+      ) : (
+        <span>{t('submitText')}</span>
+      )}
+    </Button>
+  );
+};
+
+export const ShippingInfo = ({
+  checkout,
+  shippingCountries,
+  isVisible,
+  hideShippingOptions,
+}: {
+  checkout: ExistingResultType<typeof getCheckout>;
+  shippingCountries: ExistingResultType<typeof getShippingCountries>;
+  isVisible: boolean;
+  hideShippingOptions: () => void;
+}) => {
+  const t = useTranslations('Cart.ShippingInfo');
+
+  const shippingConsignment =
+    checkout.shippingConsignments?.find((consignment) => consignment.selectedShippingOption) ||
+    checkout.shippingConsignments?.[0];
+
+  const selectedShippingCountry = shippingCountries.find(
+    (country) => country.countryCode === shippingConsignment?.address.countryCode,
+  );
+
+  const [selectedCountry, setSelectedCountry] = useState(
+    selectedShippingCountry
+      ? `${selectedShippingCountry.countryCode}-${selectedShippingCountry.id}`
+      : '',
+  );
+  const [selectedCity, setSelectedCity] = useState(shippingConsignment?.address.city || '');
+  const [selectedZipCode, setSelectedZipCode] = useState(
+    shippingConsignment?.address.postalCode || '',
+  );
+  const [selectedStates, setSelectedStates] = useState<StatesList | null>([]);
+  const [selectedState, setSelectedState] = useState(
+    shippingConsignment?.address.stateOrProvince || '',
+  );
+
+  useEffect(() => {
+    if (selectedCountry) {
+      const countryId = selectedCountry.split('-')[1];
+
+      const fetchStates = async () => {
+        const { status, data } = await getShippingStates(Number(countryId));
+
+        if (status === 'success' && data) {
+          setSelectedStates(data);
+        } else {
+          setSelectedStates(null);
+        }
+      };
+
+      if (countryId) {
+        void fetchStates();
+      }
+    }
+  }, [selectedCountry, t]);
+
+  const onSubmit = async (formData: FormData) => {
+    const { status } = await submitShippingInfo(formData, {
+      checkoutId: checkout.entityId,
+      lineItems:
+        checkout.cart?.lineItems.physicalItems.map((item) => ({
+          lineItemEntityId: item.entityId,
+          quantity: item.quantity,
+        })) || [],
+      shippingId: shippingConsignment?.entityId ?? '',
+    });
+
+    if (status === 'error') {
+      toast.error(t('errorMessage'), {
+        icon: <AlertCircle className="text-error-secondary" />,
+      });
+    }
+  };
+
+  const resetFormFieldsOnCountryChange = () => {
+    if (selectedCountry) {
+      setSelectedStates([]);
+      setSelectedState('');
+      setSelectedCity('');
+      setSelectedZipCode('');
+      hideShippingOptions();
```

Comment:
🍹 Might be a nice to refactor this as a reducer

Reply:
Updated in #711